bin/yang_get_stats.py

```diff
@@ -441,43 +72,383 @@ def write_dictionary_file_in_json(in_dict, path, file_name):
+        yang_page_compilation_stats = self._load_compilation_stats_from_history_file(json_history_file)
+        for filename in self._file_names_containing_keyword(self.files, self.IETF_YANG_OUT_OF_RFC_PREFIX):
+            rfc_result = 0
+            with open(self.backup_directory + filename, 'r') as f:
+                for line in f:
+                    if '.yang' in line:
+                        rfc_result += 1
+            extracted_date = self._extract_date_from_filename(filename)
+            if (datetime.date.today() - extracted_date).days > 30:
+                self.remove_old_html_files.append(self.backup_directory + filename)
+                yang_page_compilation_stats[date2num(extracted_date)] = {'total': rfc_result}
+            ietf_yang_out_of_rfc[date2num(extracted_date)] = {'total': rfc_result}
+        if int(args.days) == -1:
+            with open(json_history_file, 'w') as f:
+                json.dump(yang_page_compilation_stats, f)
+            self._write_dictionary_file_in_json(
+                yang_page_compilation_stats, self.stats_path, 'IETFYANGOutOfRFCStats.json'
+            )
+
+    def _load_compilation_stats_from_history_file(self, json_history_file: str) -> dict:
+        compilation_stats = {}
+        if os.path.isfile(json_history_file):
+            with open(json_history_file, 'r') as f:
+                compilation_stats_temp = json.load(f)
+            for key, value in compilation_stats_temp.items():
+                compilation_stats[float(key)] = value
+            del compilation_stats_temp
+        return compilation_stats
+
+    def _extract_date_from_filename(self, filename: str) -> datetime.date:
+        filename_without_extension = filename.split('.')[0]
+        year = filename_without_extension.split('_')[1]
+        month = filename_without_extension.split('_')[2]
+        day = filename_without_extension.split('_')[3]
+        return datetime.date(int(year), int(month), int(day))
+
+    def print_files_information(self):
+        # determine the number of company authored drafts
+        files = [
+            filename for filename in os.listdir(self.draft_path_strict) if
+            os.path.isfile(os.path.join(self.draft_path_strict, filename))
+        ]
+        files_no_strict = [
+            filename for filename in os.listdir(self.draft_path_nostrict) if
+            os.path.isfile(os.path.join(self.draft_path_nostrict, filename))
+        ]
+        total_number_drafts = len(files)
+        total_number_drafts_no_strict = len(files_no_strict)
+        print('\nPrint, per company, the number of IETF drafts containing YANG model(s)')
+        print(
+            f'Total numbers of drafts with YANG Model(s): {total_number_drafts} - '
+            f'non strict rules: {total_number_drafts_no_strict}\n'
+        )
+
+        def print_attribution(name: str, domain: str):
+            if not name and not domain:
+                print()
```

Comment:
If you put `return` after this print you can remove `else` block and unindent the code in this `else`